### DISCUSSION TITLE
Image | Radxa ROCK 5

### DISCUSSION THREAD

**Discussion by docgalaxyblock** [MichaIng/DietPi#5247] [Image Request] [answered]
#### Formal device information
- Device name | Rock 5
- Official product URL | https://wiki.radxa.com/Rock5
- GitHub resource URL (not yet avaible) | https://github.com/radxa/
- Image download URL | https://wiki.radxa.com/Rock5/downloads (not yet avaible)

#### Is the SBC officially supported by the Debian installer?
- Seems like not

#### If not, is a reliable 3rd party Debian image available for this SBC?
- Radxa should provide one at there [Wiki](https://wiki.radxa.com/Rock5/downloads)

#### My Wishes
 - Hope you can develop a good stable image like your rockpi images 😉
 - Hope you will get a dev sample eg from [here](https://shop.allnetchina.cn/collections/rock5-model-b/products/rock5-model-b), ask the shop and maybe you get one 🙏
 (seems like they have a preorder system like the steam deck https://shop.allnetchina.cn/products/rock5-model-b-rk3588-pre-order-promotion-voucher)
 
 I missed the preorder start ... hope to get one to move some homelab services.
 
 I hope the Dev team has some time 🙃
 
 Greetings from Germany 😃
_______
Vote for this image via up button at this post.

**Comment by Joulinar** [comment]
Just out of curiosity, did you tried the Zero image?

https://dietpi.com/downloads/images/DietPi_RadxaZero-ARMv8-Bullseye.7z

**Comment by MichaIng** [comment]
Jep, requires a new image, similar to Quartz64. I'll ask Radxa for a developer sample.

**Reply by isarrider** [reply]
> <img alt="" width="32" height="32" src="https://avatars.githubusercontent.com/u/28480705?s=64&amp;v=4">
> MichaIng

did you get a reply - I am still waiting for my preorder to be able to ship...

**Reply by MichaIng** [reply]
I just sent out the request, along with one for ROCK3 (Model A) and the outstanding Radxa Zero shipment.

**Reply by docgalaxyblock** [reply]
Did you received a sample yet?

**Reply by isarrider** [reply]
I wish - but sadly no...

**Reply by docgalaxyblock** [reply]
The DEV samples are finally there ! 😍
https://forum.radxa.com/t/rock-5b-debug-party-invitation/10483
@MichaIng Ask again for every Radxa sample you did before

**Comment by MichaIng** [comment]
Converted into discussion, so here votes can be placed.

**Comment by isarrider** [comment]
it seems the boards are starting to ship,
so is there an update?

**Reply by docgalaxyblock** [reply]
AFAIK (as a user) they hope to get all preorders processed this month.
@MichaIng Did you had success regarding a dev sample? (Also the other radxa board?)

I have preordered in February and have not received a code email yet.
🤞

**Reply by isarrider** [reply]
I preordered in Jan and my board seems to ship soon...

**Reply by MichaIng** [reply]
Sadly no feedback. Armbian has unofficial community releases in the meantime, sadly for Debian Sid (unstable) only.

**Comment by docgalaxyblock** [comment]
I just got the order link emailed by allnetchina 🥳

**Reply by docgalaxyblock** [reply]
It arrived today - it took just one week to germany
https://cdn.discordapp.com/attachments/494242871583571971/1034755314207162399/IMG_20221026_110712.jpg
https://cdn.discordapp.com/attachments/494242871583571971/1034755798426996736/IMG_20221026_110952.jpg

**Comment by lukaszsobala** [comment]
There is a Rock5 Debian image available on their website (as the OP wrote)

**Reply by MichaIng** [reply]
Also Armbian legacy (vendor kernel/u-boot) is available now. But I'll wait for a mainline kernel image before I integrate it into DietPi.

**Comment by RainmakerRaw** [comment]
Watching this with interest. My Rock 5 model B 16GB arrived today. I'll flash their Debian Bullseye image for now and just strip out the X and GUI stuff, but the moment DietPi is out it'll be getting re-flashed!

**Comment by MichaIng** [comment, marked as answer]
PR up to add support: https://github.com/MichaIng/DietPi/pull/5917
Test image builds are running.

There is one issue with device tree overlays (for enabling I2C/SPI/UART, 8k support for HDMI and such): The boot script uses a (file name) prefix to load (only matching) overlay files, but the ones provided by Armbian's current RK3588 kernel package have several different prefixes valid for ROCK 5B. So depending on which one is provided in `dietpiEnv.txt`, either only one set or another set of overlays will work, and the fixup to repair after applying an overlay failed won't work at all, as it has an individual prefix 😄. Will be fixed earlier or later, but until then, we need to use a highly customised `boot.cmd`/`boot.scr` which doesn't use prefixes at all so that you can enter it freely into the overlays setting in `dietpiEnv.txt`.

**Reply by MichaIng** [reply]
Here they are: https://dietpi.com/downloads/images/

**Reply by lukaszsobala** [reply]
I might try them soon if I manage to boot the thing... On its way now

**Reply by RainmakerRaw** [reply]
I still hadn't gotten around to booting mine. I'll re-flash the eMMC to this test image (Bullseye for me) and let you know how it goes. Thanks for all the hard work!

**Reply by docgalaxyblock** [reply]
The Bullseye Image boots and I did a dietpi-benchmark
~10:25 UTS+2 should be the upload time as a generic device

~10:36 i tried to install a desktop but it has some problem regarding alsa
bug report reference is `fca71361-...`
workaround by replacing `alsactl -g store` by another command and the desktop installs
Now I need to go

**Reply by MichaIng** [reply]
Many thanks! Ah, makes sense to recreate those images with `dev` branch, so the SBC name is correctly detected. You can switch easily:
```sh
G_DEV_BRANCH dev
```

Mind-blowing:
```sh
BENCH_CPU='1.87'
BENCH_RAM_WRITE='1023'
BENCH_RAM_READ='3684'
BENCH_CPU_TEMP_START='32'
BENCH_CPU_TEMP_END='36'
```
The fastest ever measured SBC until now was Odroid N2+ with `3.56` (average for Odroid N2 is `4.01`) seconds: https://dietpi.com/survey/#benchmark
So this one has more than doubled CPU power. RAM speed however is on same level with Odroid N2, both with some distance fastest SBCs in our RAM benchmarks: https://dietpi.com/survey/#bench_ram

> i tried to install a desktop but it has some problem regarding alsa

So HDMI audio doesn't seem to work OOTB, I guess. What does this show:
```sh
aplay -l
```
Ah, from the bug report upload:
```
**** List of PLAYBACK Hardware Devices ****
card 0: rockchiphdmi0 [rockchip-hdmi0], device 0: rockchip-hdmi0 i2s-hifi-0 [rockchip-hdmi0 i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: rockchiphdmi1 [rockchip-hdmi1], device 0: rockchip-hdmi1 i2s-hifi-0 [rockchip-hdmi1 i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: rockchipes8316 [rockchip-es8316], device 0: fe470000.i2s-ES8316 HiFi es8316.7-0011-0 [fe470000.i2s-ES8316 HiFi es8316.7-0011-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```
```
alsactl: add_tlv_comments:216: snd_tlv_convert_to_dB: Invalid argument
```
So somehow the default states cannot be stored:
```sh
alsactl init # should reset ALSA states to a valid default
alsactl -g store # should store those defaults to ALSA states file, but fails
```
Similar to this issue on RPi: #4306
We still work around this by ignoring the error in case of RPi. Not sure if still required, or whether we probably should ignore it in all cases. `amixer` would fail directly if we tried to add invalid arguments ourselves, and `alsamixer` run by user likely solves the issue, and states are stored automatically on regular shutdown, so `alsactl -g store` is only to force it in case or crash or such, to assure a valid default state on next boot. If `alsactl init` does not init a valid default state, this is redundant anyway.

**Reply by MichaIng** [reply]
It is now generally ignored if this command fails. It however means, if it keeps failing, that `amixer`/`alsamixer` changes do not survive reboots, so in this case it would need to be further investigated: https://github.com/MichaIng/DietPi/commit/e616453

**Reply by MrHoozer** [reply]
Hello MichaIng,

I know this is already "very old", but I ran into the issue that it still seems impossible to me to enable I2C-functionality on my Rock5B using "DietPi v9.9.0".
I searched through the options of the "dietpi-config", "-launcher" and also "-software" multiple times in the hope to find the menuu to enable I2C in an "easy way", but there are no menu-options to do it.
Then I tried it with an entry into the "dietpiEnv.txt", but obviously also with no visible success.

During my internet-search for solutions I came across an other post in the "dietpi.com"-forum that is from 04.2023 and was never answered:
[https://dietpi.com/forum/t/rock5b-lcd-not-found-on-i2c-7/16484](url)
But it is describing exactly what I tried (except that I used "overlays=rk3588-i2c7-m3") to get my "MPU-9255" running on a Rock5B via "IC27_M3"-connection to use it in "Klipper" for input-shaping.

I also came across an other post somewhere where the enabling of the I2C worked with an other linux-distro by editing the "ubuntuEnv.txt":
[https://ameblo.jp/kissam59/entry-12832900363.html](url)
So it should be possible somehow to get it working.

So it seems to me like you described in your quoted answer below that this function still does not work and it is impossible to use I2C or any other "overlay" to enable features of the Rock5B:
> PR up to add support: #5917 Test image builds are running.
> 
> There is one issue with device tree overlays (for enabling I2C/SPI/UART, 8k support for HDMI and such): The boot script uses a (file name) prefix to load (only matching) overlay files, but the ones provided by Armbian's current RK3588 kernel package have several different prefixes valid for ROCK 5B. So depending on which one is provided in `dietpiEnv.txt`, either only one set or another set of overlays will work, and the fixup to repair after applying an overlay failed won't work at all, as it has an individual prefix 😄. Will be fixed earlier or later, but until then, we need to use a highly customised `boot.cmd`/`boot.scr` which doesn't use prefixes at all so that you can enter it freely into the overlays setting in `dietpiEnv.txt`.

**So my question is:**
**Is it possible to use I2C7_M3 (or any other "overlay"-feature/-connection) and if so, is there a step-by-step-manual to accomplish this?**

I am not a Linux-Pro, so my knowledge is just a bunch of collected infos I came across during my desperate search for a solution.

It would be realy disappointing that all these feature-possibilities would be left unuseable in "DietPi".


I realy hope to get some new infos about this topic and maybe even a solution I can use to take full advantage of my RockB and the very nice "DietPi"!

Thanks in advance (and a Happy New Year!)

Hoozer

**Reply by MichaIng** [reply]
Set `overlays=i2c7-m3`. The `overlay_prefix=rk3588` is what is removed from the filename (plus the dash), when the boot script looks for matching overlays.

And please open a dedicated issue next time. Appended to a years old image request, such question is hard to be found by anyone else 😉.

**Comment by isarrider** [comment]
I am also playing around with it now - thank you...
Docker wont install - error log 2baa9a1b-e962-4f0f-8193-cb02241c112e
I am checking currently - I remember that with the Armbian Image
I needed to change sth too... (but cant remember clearly)...

**Reply by MichaIng** [reply]
`docker_optimizations=on` should be set in `/boot/dietpiEnv.txt` automatically when installing Docker via `dietpi-software`. Another question is whether cgroupv2 and BPF support is added with the kernel. Usually it is on Armbian images. However, e.g. on ASUS Tinker Board cgroupv2 must be disabled actively to work with Docker, and on Quartz64 related kernel modules are not enabled (same solution, but that's no Armbian kernel): #5890

**Reply by isarrider** [reply]
cgroupv2 is enabled:
"grep cgroup /proc/filesystems
nodev   cgroup
nodev   cgroup2"

**Reply by isarrider** [reply]
seems BPF is also, as /proc/config.gz menations it as yes...

**Reply by MichaIng** [reply]
It always is by default since Bullseye. The question is whether the kernel fully supports it, or whether Docker fails with `bpf_prog_query(BPF_CGROUP_DEVICE) failed: function not implemented`, in which case cgroupv2 would need to be disabled:
```sh
grep -q '^extraargs=.*systemd.unified_cgroup_hierarchy=0' /boot/dietpiEnv.txt || sed -i '/^extraargs=/s/$/ systemd.unified_cgroup_hierarchy=0/' /boot/dietpiEnv.txt
```

EDIT:

> seems BPF is also, as /proc/config.gz menations it as yes...

On ASUS Tinker Board it is as well, but it still fails with this error and the reason has not been found in months.

**Reply by isarrider** [reply]
shouldnt this be: "grep -q '^extraargs=.*systemd.unified_cgroup_hierarchy=0' /boot/dietpiEnv.txt || sed -i '/^extraargs=/s/$/ systemd.unified_cgroup_hierarchy=0/' /boot/dietpiEnv.txt" ?

**Reply by isarrider** [reply]
Either way, I ran docker in debug mode and one of the key lines it spit out was:
WARN[2022-11-25T14:27:28.010117015Z] Running iptables --wait -t nat -L -n failed with message: `iptables v1.8.7 (nf_tables): Could not fetch rule set generation id: Invalid argument`, error: exit status 4
That can then be resolved by:
update-alternatives --set iptables /usr/sbin/iptables-legacy
( I think I did the same on the old armbian image)...

**Reply by MichaIng** [reply]
Ah right, of course.

**Reply by MichaIng** [reply]
Interesting, that kernel should definitely support nftables, also otherwise `dietpi-software` would have applied legacy alternative automatically. But probably some nftables sub feature is missing or so.

**Reply by isarrider** [reply]
shall I open an issue or only when this becomes "stable"?

**Reply by MichaIng** [reply]
Let's finish these device specific things here.

**Reply by lukaszsobala** [reply]
I tried to install ALSA but it failed - there should be a report from an hour or so ago (EDIT: I can see this was spotted earlier). Many other things installed correctly.

Another problem I saw were artifacts when playing a YouTube video. I haven't changed any graphical options or played with drivers yet, although I intend to.

Also the fan doesn't spin but I know it's plugged in correctly (but I assume I need a deb from radxa). 

One more thing - I will need to investigate if WiFi indeed works, as it seems connected but may be intermittent. Ethernet works  great (>900 Mbps on a Gbit connection).

**Reply by MichaIng** [reply]
As well with `alsactl -g store` as reported [above](https://github.com/MichaIng/DietPi/discussions/5247#discussioncomment-4236146)?

**Comment by MrHoozer** [comment]
Hello MichaIng,one last question:At what BoudRate I2C is set by default?Is there also a way to specify that via the "...Env"-file?Thanks a lot for your great supportAndreasAm 04.01.25, 09:39 schrieb MichaIng ***@***.***>:

Set overlays=i2c7-m3. The overlay_prefix=rk3588 is what is removed from the filename (plus the dash), when the boot script looks for matching overlays.
And please open a dedicated issue next time. Appended to a years old image request, such question is hard to be found by anyone else 😉.

**Reply by MichaIng** [reply]
No you cannot pass parameters to these overlays. About the baudrate: check the SBC wiki/scheme: https://wiki.radxa.com/Rock5/hardware

Or the device tree:
- https://github.com/armbian/linux-rockchip/blob/rk-6.1-rkr4.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts#L1049-L1061
    - https://github.com/armbian/linux-rockchip/blob/rk-6.1-rkr4.1/arch/arm64/boot/dts/rockchip/rk3588s.dtsi#L6616-L6629
        - ... okay it does not seem to be defined/hardcoded there, but maybe derived from the hardware directly.

I would assume 100 kbit/s.
